cmd/docker/docker.go

```diff
@@ -227,9 +234,14 @@ func runDocker(dockerCli *command.DockerCli) error {
 		return err
 	}
 
-	err = pluginmanager.AddPluginCommandStubs(dockerCli, cmd)
-	if err != nil {
-		return err
+	if cli.HasCompletionArg(args) {
+		// We add plugin command stubs early only for completion. We don't
+		// want to add them for normal command execution as it would cause
+		// a significant performance hit.
+		err = pluginmanager.AddPluginCommandStubs(dockerCli, cmd)
```

Comment:
Even if there is a completion arg, why do we need to list all the plugins? Most  likely the command is only for a specific plugin. Asking for follow-up, not a blocker.

Reply:
Because we need to list completion values for plugins too so they have to be registered first.

Reply:
Only when the args is `docker`. Not when it is already `docker buildx`. This is not that important as only affects the completion path but ideally it shouldn't be that adding more plugins makes completion for another plugin slower.

Reply:
Yes indeed, we could improve this.